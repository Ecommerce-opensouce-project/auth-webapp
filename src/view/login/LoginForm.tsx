import * as React from "react";
import { Button } from 'ecommerce-opensouce-storybook'

const LoginForm = () => <div>
    <div className="login-name">Name</div>
    <div className="login-password">Password</div>
    <Button label="Login"/>
    <Button label="Register"/>
</div>


export default LoginForm;