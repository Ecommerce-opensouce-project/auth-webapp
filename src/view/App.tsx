import * as React from "react";
import LoginForm  from "./login/LoginForm";

export default () => (
  <div>
    <div className="card-header">Welcome</div>
    <div className="container">
      <LoginForm/>
    </div>
  </div>
); 