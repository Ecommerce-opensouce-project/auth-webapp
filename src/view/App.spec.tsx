import * as React from 'react';
import { shallow } from "enzyme";
import * as renderer from 'react-test-renderer';
import App  from "./App";
import LoginForm from './login/LoginForm';

const wrapper = shallow(<App/>);

describe('Test Auth Webapp entry', function () {
    it('test againest snapshot', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should have a header tag with Welcome', function () {
        expect(wrapper.find(".card-header").text()).toEqual("Welcome");
    });
    it('should include loginform by default', function () {
        expect(wrapper.find(LoginForm).exists()).toBe(true);
    });
});