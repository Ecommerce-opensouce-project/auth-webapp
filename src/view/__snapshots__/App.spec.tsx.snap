// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Auth Webapp entry test againest snapshot 1`] = `
<div>
  <div
    className="card-header"
  >
    Welcome
  </div>
  <div
    className="container"
  >
    <div>
      <div
        className="login-name"
      >
        Name
      </div>
      <div
        className="login-password"
      >
        Password
      </div>
      <button
        className="storybook-button storybook-button--medium storybook-button--secondary"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        type="button"
      >
        Login
      </button>
      <button
        className="storybook-button storybook-button--medium storybook-button--secondary"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        type="button"
      >
        Register
      </button>
    </div>
  </div>
</div>
`;
